cmd/kubeadm/app/phases/kubeconfig/kubeconfig.go

```diff
@@ -101,17 +83,13 @@ func createKubeConfigFileForClient(masterEndpoint, client, outDir string, caCert
 }
 
 func WriteKubeconfigToDisk(filepath string, kubeconfig *clientcmdapi.Config) error {
-	// Make sure the dir exists or can be created
-	if err := os.MkdirAll(path.Dir(filepath), KubernetesDirPermissions); err != nil {
-		return fmt.Errorf("failed to create directory %q [%v]", path.Dir(filepath), err)
-	}
-
 	// If err == nil, the file exists. Oops, we don't allow the file to exist already, fail.
+	// TODO: Should we allow overwriting a kubeconfig file that does already exist?
```

Comment:
What about if we first try to create the kubeconfig files, make a bytearray of them and compare them to the existing ones?

Then it would fail if a file exist that not matches the expected content, but just continue (skip) if a file with that name and content already exists.
WDYT?